lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -740,40 +743,56 @@ monitorMetadata gcStatus tr sp db@(DBLayer{..}) = do
     settings <- atomically readSettings
     manager <- newManager defaultManagerSettings
 
-    let fetcher fetchStrategies = fetchFromRemote trFetch fetchStrategies manager
-        loop getPoolMetadata = forever $ do
-            (refs, successes) <- getPoolMetadata
-            when (null refs || null successes) $ do
-                traceWith tr $ MsgFetchTakeBreak blockFrequency
-                threadDelay blockFrequency
-
-    case poolMetadataSource settings of
-        FetchNone -> do
-            STM.atomically $ writeTVar gcStatus NotApplicable
-            loop (pure ([], [])) -- TODO: exit loop?
-        FetchDirect -> do
-            STM.atomically $ writeTVar gcStatus NotApplicable
-            loop (fetchThem $ fetcher [identityUrlBuilder])
-        FetchSMASH (unSmashServer -> uri) -> do
-            STM.atomically $ writeTVar gcStatus NotStarted
-            let getDelistedPools =
-                    fetchDelistedPools trFetch (toDelistedPoolsURI uri) manager
-            tid <- forkFinally
-                (gcDelistedPools gcStatus tr db getDelistedPools)
-                onExit
-            flip finally (killThread tid) $
-                loop (fetchThem $ fetcher [registryUrlBuilder uri])
+    health <- case poolMetadataSource settings of
+        FetchSMASH uri -> do
+            let checkHealth _ = do
+                    r <- healthCheck trFetch (unSmashServer uri) manager
+                    case r of
+                        (Just health)
+                            | isHealthyStatus health -> pure Available
```

Comment:
We are doing this transformation in two places, which comfort me with the idea that `healthCheck` should really return a `HealthCheckSMASH` datatype in the first place instead of a `Maybe Bool`.